src/Parser.zig

```diff
@@ -136,21 +136,36 @@ fn checkIdentifierCodepoint(comp: *Compilation, codepoint: u21, loc: Source.Loca
 }
 
 fn eatIdentifier(p: *Parser) !?TokenIndex {
-    if (p.tok_ids[p.tok_i] == .identifier) {
-        defer p.tok_i += 1;
-        return p.tok_i;
-    } else if (p.tok_ids[p.tok_i] == .extended_identifier) {
-        defer p.tok_i += 1;
-        const slice = p.tokSlice(p.tok_i);
-        var it = std.unicode.Utf8View.initUnchecked(slice).iterator();
-        var loc = p.pp.tokens.items(.loc)[p.tok_i];
-        while (it.nextCodepoint()) |c| {
-            if (try checkIdentifierCodepoint(p.pp.comp, c, loc)) break;
-            loc.byte_offset += std.unicode.utf8CodepointSequenceLength(c) catch unreachable;
+    switch (p.tok_ids[p.tok_i]) {
+        .identifier => {
+            defer p.tok_i += 1;
+        },
```

Comment:
```suggestion
        .identifier => {},
```
The `defer p.tok_i += 1;` can be added to the end of the function.